samples/secure-vertex-ai-gemini-poem/README.md

```diff
@@ -0,0 +1,111 @@
+# Secure Vertex AI Gemini Poem Demo
+
+This demo showcases the implementation of a secure Vertex AI Gemini Poem Demo which is available only to users authenticated with Google.
+
+## The Demo
+
+Demo asks Vertex AI Gemini LLM to write a short 1 paragraph poem, using the access token acquired during the OIDC authorization code flow
+
+### Setup
+
+This demo requires users to authenticate with Google.
+All you need to do is to register an application with Google, follow steps listed in the [Quarkus Google](https://quarkus.io/guides/security-openid-connect-providers#google) section.
+Name your Google application as `Quarkus LangChain4j AI`, and make sure an allowed callback URL is set to `http://localhost:8080/login`.
+Google will generate a client id and secret, use them to set `quarkus.oidc.client-id` and `quarkus.oidc.credentials.secret` properties:
+
+```properties
+quarkus.oidc.provider=google
+quarkus.oidc.client-id=${GOOGLE_CLIENT_ID}
+quarkus.oidc.credentials.secret=${GOOGLE_CLIENT_SECRET}
+quarkus.oidc.authentication.extra-params.scope=https://www.googleapis.com/auth/generative-language.retriever,https://www.googleapis.com/auth/cloud-platform
+quarkus.oidc.authentication.redirect-path=/login
+
+# See https://cloud.google.com/vertex-ai/docs/general/locations
+vertex-ai-region=europe-west2
+
+quarkus.langchain4j.vertexai.gemini.location=https://${vertex-ai-region}-aiplatform.googleapis.com
+quarkus.langchain4j.vertexai.gemini.project-id=${GOOGLE_PROJECT_ID}
```

Comment:
Please explain what `GOOGLE_PROJECT_ID` is and how it differs from `GOOGLE_CLIENT_ID`

Reply:
@jmartisk I updated the Setup section to explain what Google Project ID is